### PR TITLE
DC-415: link to workspace support; add workspace URL to dataset metadata

### PR DESCRIPTION
**Pull request by pshapiro4broad**
Updated workspace generated catalog metadata to include `accessURL`, which contains the link to the corresponding terra workspace.

Also updated the HCA script generated accessURL, and other minor cleanups to the HCA code.

### CODE REVIEW

File: scripts/workspace-conversion/workspace-to-dataset.py [hunk 1]
```diff
@@ -266,6 +267,19 @@ def get_workspace_files(wsAttributes):
     return ret
 
 
+def access_url(workspace):
+    p = re.compile("rawls.dsde-(\\w+)")
+    env = p.search(urlRoot).group(1)
+    if env == "prod":
+        terra_url = "https://app.terra.bio"
+    else:
+        terra_url = f"https://bvdp-saturn-{env}.appspot.com"
```
Comment (fboulnois) [inline scripts/workspace-conversion/workspace-to-dataset.py:276]:
```suggestion
    terra_url = "https://app.terra.bio" if env == "prod"
    else f"https://bvdp-saturn-{env}.appspot.com"
```

**Review by fboulnois — COMMENTED** [review]
lgtm, one minor suggestion:

File: scripts/workspace-conversion/workspace-to-dataset.py (hunk 1 above)
Comment (pshapiro4broad) [reply scripts/workspace-conversion/workspace-to-dataset.py]: FYI this makes the lines too long, so after `black` reformatting I get
```
    terra_url = (
        "https://app.terra.bio"
        if env == "prod"
        else f"https://bvdp-saturn-{env}.appspot.com"
    )
```
not sure if this is better / worse than the original `if` at this point

**Review by s-rubenstein — APPROVED** [review]
LGTM

File: scripts/workspace-conversion/README.md [hunk 2]
```diff
@@ -21,41 +21,47 @@ The request made to retrieve the workspace details is an XHR request to rawls to
...
-| 22 | Study Design | `library:studyDesign`<br>ex: "Tumor/Normal" | `samples.disease.0`<br>Note: concat value with "Primary Disease Site"? |
-| 23 | Cell Type | `library:cellType`<br>ex: "Primary tumor cell, Whole blood" | `samples.disease.0`<br>Note: concat value with "Primary Disease Site"? |
-| 24 | Reported Ethnicity | `library:ethnicity` |  |
-| 25 | Cohort Country of Origin | `library:cohortCountry`<br>ex: "USA" |  |
-| 26 | Requires External Approval | `library:requiresExternalApproval` |  |
-| 27 | library Metadata Schema Version Number | `library:lmsvn` |  |
-| 28 | Structured Data Use Limitations Version Number | `library:dulvn` |  |
-| 29 | ORSP Conset Code | `library:orsp` | `TerraDCAT_ap:hasConsentGroup` |
+| i   | Attribute                                      | Rawls Workspace Path | Data Catalog Entry Path |
+|-----|------------------------------------------------| -------------------- | -------------------------- |
```
Comment (pshapiro4broad) [inline scripts/workspace-conversion/README.md:32]: This is an auto-reformat by IntelliJ, which happened once I updated the header line for `i` to have three hyphens, which is what the markdown spec requires for a column header row.

**Review by fboulnois — APPROVED** [review]

File: scripts/workspace-conversion/workspace-to-dataset.py (hunk 1 above)
Comment (fboulnois) [reply scripts/workspace-conversion/workspace-to-dataset.py]: Yeah, you're right, I'm not sure now.

**Pull request merged**